pdf_parsing/parser.py

```diff
@@ -2,102 +2,152 @@
-
-
-def parse_page_text(text: str):
-    sentences = text.replace("\t", " ").split("\n")
-
-    quote = ""
-    explanation = ""
-    end_quote_symbol = "”"
-    start_source_symbol = "—"
-    quote_found = False
-    quote_source_found = False
-
-    if len(sentences[0]) == 1:
-        for index, sentence in enumerate(sentences):
-            if index == 0:
-                missing_char = sentence
-            elif index == 1:
-                date = sentence[:-2]
-            elif index == 2:
-                title = sentence
-            else:
-                if not quote_found:
-                    if sentence[0] == start_source_symbol:
-                        quote_found = True
-                        quote_source = sentence
-                        quote_source_found = True
-
-                    if sentence[-1] != end_quote_symbol:
-                        quote += sentence
-                    else:
-                        quote += sentence
-                        quote_found = True
-                else:
-                    if not quote_source_found:
-                        quote_source = sentence
-                        quote_source_found = True
-                        explanation += missing_char
-                    else:
-                        explanation += sentence
-    else:
-        for index, sentence in enumerate(sentences):
-            if index == 0:
-                date = sentence[:-2]
-            elif index == 1:
-                title = sentence
-            else:
-                if not quote_found:
-                    if sentence[-1] != end_quote_symbol:
-                        quote += sentence
-                    else:
-                        quote += sentence
-                        quote_found = True
-                else:
-                    if not quote_source_found:
-                        quote_source = sentence
-                        quote_source_found = True
-                    else:
-                        explanation += sentence
-
-    return date, title, quote, quote_source, explanation
-
-
-def extract_text(pdf_file, start, end, breaks):
-    entries = {}
-
-    with fitz.open(pdf_file) as f:
-        for page in f:
-            if page.number < start:
+PARSING_FOLDER = "./pdf_parsing/"
+RESULT_FILE_PATH = PARSING_FOLDER + "Stoic_log.json"
+
+# These needs to be set according to a specific copy of The Daily Stoic book.
+# Page numbers start from 0, so page number in PDF minus 1
+FIRST_STOIC_PAGE = 14
+LAST_STOIC_PAGE = 392
+SKIP_NON_STOIC_PAGES = [
+    45,
+    75,
+    107,
+    138,
+    139,
+    171,
+    202,
+    234,
+    266,
+    267,
+    298,
+    330,
+    361,
+]
+START_QUOTE_SYMBOL = "\u201c"  # Origin: “
+END_QUOTE_SYMBOL = "\u201d"  # Origin: ”
+START_SOURCE_SYMBOL = "\u2014"  # Origin: —
+
+
+# Parses The Daily Stoic book.
+# The book is a PDF file with a specific structure.
+class Parser:
+    # Returns a list of all pdf files in a folder.
```

Comment:
this comment seems redundant and/or you could make the method name below more descriptive